python/ray/services.py

```diff
@@ -505,46 +445,89 @@ def start_redis(node_ip_address,
         raise Exception("The number of Redis shard ports does not match the "
                         "number of Redis shards.")
 
-    assigned_port, _ = start_redis_instance(
-        node_ip_address=node_ip_address, port=port,
-        redis_max_clients=redis_max_clients,
-        stdout_file=redis_stdout_file, stderr_file=redis_stderr_file,
-        cleanup=cleanup)
+    if no_credis:
+        assigned_port, _ = start_redis_instance(
+            node_ip_address=node_ip_address,
+            port=port,
+            redis_max_clients=redis_max_clients,
+            stdout_file=redis_stdout_file,
+            stderr_file=redis_stderr_file,
+            cleanup=cleanup)
+    else:
+        # It is import to load the credis module BEFORE the ray module, as the
```

Comment:
Is the relevant line for this at 467? If yes, move the comment down to that line (also "import" --> "important").